GithubPractice/ViewController.swift

```diff
@@ -59,6 +63,40 @@ class ViewController: UIViewController {
             defaults.removeObject(forKey: key)
         }
     }
+    
+    func saveData(){
+        saveName()
+        defaults.set(lastNameField.text, forKey: Constants.lastNameKey)
+        defaults.set(jobPositionField.text, forKey: Constants.jobKey)
+        defaults.set(birthdayPicker.date, forKey: Constants.birthDayKey)
+    }
+    
+    func saveName(){
+        if let firstName = firstNameField.text {
+            if validation(str: firstName) {
+                saveResult = .succes
+            } else {
+                saveResult = .failure
+            }
+            
+            switch saveResult {
+            case .succes:
+                saveButton.backgroundColor = .green
+                defaults.set(firstNameField.text, forKey: Constants.firstNameKey)
+            case .failure:
+                saveButton.backgroundColor = .red
+            }
+        }
+    }
+    
+    func validation(str: String) -> Bool{
+        let components = str.components(separatedBy: .whitespacesAndNewlines)
+        if components.count == 1 {
+            return true
+        } else {
+            return false
+        }
```

Comment:
use ternary

Reply:
this could be extracted as String extension. otherwise there is no reason to return value, you could set saveResult value right here

Reply:
done